### PR TITLE
reboot: Add a simple plugin to reboot a machine

### PR DESCRIPTION
**Pull request by neverpanic**
Assumes systemd, beacuse it uses systemctl to wait for the boot to complete.

### CODE REVIEW

**Comment by t184256** [comment]
> reboot: Add a simple plugin to reboot a machine

Good idea, this is a common scenario.

> Assumes systemd, beacuse it uses systemctl to wait for the boot to complete.

Yeah, since this the step is distro-specific, this should be made more flexible with hooks.
See, for example, how timesync decouples this.

**Comment by neverpanic** [comment]
Untested with `os.alpine`, because that seems to be broken.
Also, `os.alpine` does not define `m.login()`, so the reboot action is also broken because of that. I may eventually fix the `os.alpine` definition, but quite a few things seem to have changed in alpine land, so it isn't trivial.

File: fingertip/plugins/reboot.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Licensed under GNU General Public License v3 or later, see COPYING.
+# Copyright (c) 2024 Red Hat, Inc., see CONTRIBUTORS.
+
+def main(m):
+    with m:
```
Comment (t184256) [inline fingertip/plugins/reboot.py:5]: too defensive for fingertip =) just fire-and-forget, worst case it'd get snapshotted mid-boot

File: fingertip/util/systemd.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+"""
```
Comment (t184256) [inline fingertip/util/systemd.py:1]: an entire file for what's a one-liner lambda? nope, please just inline into corresponding os plugins. also, if it were larger, it still wouldn't belong to `utils/`, but rather to `plugins/os/common`

Comment (neverpanic) [reply fingertip/util/systemd.py:1]: I can make it `fingertip/util/initsystems.py` and include the oneliners for OpenRC and Upstart there, if you want?

I could put it into `plugins/os/common`, but then I can't use those from the RHEL OS plugins, can I? Those would also have to duplicate the systemd oneliner, which brings us to 6 copies of the same line (RHEL 7 uses systemd, but needs a different line), which I think is enough to refactor into a separate file.

If you still disagree, I can copy/paste it into all those files, but that feels dirty.

Comment (neverpanic) [reply fingertip/util/systemd.py:1]: I believe this should now be resolved, PTAL.

File: fingertip/plugins/reboot.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Licensed under GNU General Public License v3 or later, see COPYING.
+# Copyright (c) 2024 Red Hat, Inc., see CONTRIBUTORS.
+
+def main(m):
+    if 'wait_for_running' not in m.hooks:
+        m.log.critical(
```
Comment (t184256) [inline fingertip/plugins/reboot.py:6]: Please downgrade that to a `.warning()` and drop the RuntimeError. I don't see how not having it is anything more than a missed performance opportunity.

Comment (neverpanic) [reply fingertip/plugins/reboot.py]: Done.

**Comment by t184256** [comment]
Thank you, doubly so for patience with my review.

**Pull request merged**